src/vmm/src/vstate/memory.rs

```diff
@@ -543,67 +521,50 @@ mod tests {
+        let write_mem = |addr, val| unsafe {
+            std::ptr::write(addr, val);
+        };
+
         let page_size = get_page_size().unwrap();
 
         // Check that the created range allows us to write inside it
-        let addr = region.as_ptr();
+        let region_first_byte = region.as_ptr();
+        let region_last_byte = unsafe { region_first_byte.add(region.size() - 1) };
+        // let region_size = region.size();
 
-        unsafe {
-            std::ptr::write(addr, 0xFF);
-            assert_eq!(std::ptr::read(addr), 0xFF);
-        }
+        // Write and read from the start of the region
+        write_mem(region_first_byte, 0x69);
+        assert_eq!(read_mem(region_first_byte), 0x69);
+
+        // Write and read from the end of the region
+        write_mem(region_last_byte, 0x69);
+        assert_eq!(read_mem(region_last_byte), 0x69);
 
         // Try a read/write operation against the left guard border of the range
-        let left_border = (addr as usize - page_size) as *mut u8;
-        fork_and_run(&|| AddrOp::Read.apply_on_addr(left_border), true);
-        fork_and_run(&|| AddrOp::Write.apply_on_addr(left_border), true);
+        let left_border_first_byte = unsafe { region_first_byte.sub(page_size) };
+        fork_and_run(&|| write_mem(left_border_first_byte, 0x69), true);
+        fork_and_run(&|| _ = read_mem(left_border_first_byte), true);
 
         // Try a read/write operation against the right guard border of the range
-        let right_border = (addr as usize + region.size()) as *mut u8;
-        fork_and_run(&|| AddrOp::Read.apply_on_addr(right_border), true);
-        fork_and_run(&|| AddrOp::Write.apply_on_addr(right_border), true);
-    }
-
-    fn loop_guard_region_to_sigsegv(region: &GuestMmapRegion) {
-        let page_size = get_page_size().unwrap();
-        let right_page_guard = region.as_ptr() as usize + region.size();
-
-        fork_and_run(
-            &|| {
-                let mut addr = region.as_ptr() as usize;
-                loop {
-                    if addr >= right_page_guard {
-                        break;
-                    }
-                    AddrOp::Write.apply_on_addr(addr as *mut u8);
-
-                    addr += page_size;
-                }
-            },
-            false,
-        );
-
-        fork_and_run(
-            &|| {
-                AddrOp::Write.apply_on_addr(right_page_guard as *mut u8);
-            },
-            true,
-        );
+        let right_border_first_byte = unsafe { region_last_byte.add(1) };
+        fork_and_run(&|| write_mem(right_border_first_byte, 0x69), true);
+        fork_and_run(&|| _ = read_mem(right_border_first_byte), true);
     }
 
     #[test]
     fn test_build_guarded_region() {
         // Create anonymous guarded region.
         {
             let page_size = get_page_size().unwrap();
-            let size = page_size * 10;
+            let region_size = page_size * 10;
```

Comment:
since you're at it, how about splitting this test into 2?

Reply:
done